spec.html

```diff
@@ -8639,7 +8639,16 @@
           1. Return ? _target_.[[Enumerate]]().
         1. Let _trapResult_ be ? Call(_trap_, _handler_, &laquo; _target_ &raquo;).
         1. If Type(_trapResult_) is not Object, throw a *TypeError* exception.
-        1. Return _trapResult_.
+        1. Let _array_ be ArrayCreate(0).
+        1. Let _nextIndex_ be 0.
+        1. Repeat
+          1. Let _next_ be ? IteratorStep(_trapResult_).
+          1. If _next_ is *false*, return CreateArrayIterator(_array_, `"value"`).
+          1. Let _nextValue_ be ? IteratorValue(_next_).
+          1. If Type(_nextValue_) is not String, throw a *TypeError* exception.
+          1. Let _status_ be ? CreateDataProperty(_array_, ToString(ToUint32(_nextIndex_)), _nextValue_).
```

Comment:
Since CreateDataProperty can't throw, I'd say replace ? with !. It might also be more clear to say: `Perform ! CreateDataPropertyOrThrow` and remove the next line.